sync/rpc/server.go

```diff
@@ -282,7 +326,23 @@ func (*SyncAgentServer) BackupRemove(ctx context.Context, req *BackupRemoveReque
 	return &Empty{}, nil
 }
 
-func (*SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (*Empty, error) {
+func (s *SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (*Empty, error) {
+	if strings.HasSuffix(req.SnapshotFileName, ".snap_tmp") {
```

Comment:
Is this for incremental restore fall back? I don't think we need to check this since normal full restore should able to update `restoring` and `lastRestored` as well.

Reply:
Yes, this is for incremental restore fall back. The normal full restore wasn't updating `restoring` and `lastRestored` even earlier. These two fields are exclusively for incremental restore. If the incremental restore is successful, we are updating `lastRestored` and `isRestoring` below. Otherwise, we are only updating `isRestoring`, just like how it was done before.

Reply:
Unaddressed?